packages/issuer/lib/VcIssuer.ts

```diff
@@ -82,7 +85,33 @@ export class VcIssuer {
     throw new Error(TokenErrorResponse.invalid_request)
   }
 
-  private async validateJWT(issueCredentialRequest: CredentialRequest, jwtVerifyCallback?: JWTVerifyCallback, clientId?: string, preAuthorizedCode?: string): Promise<void> {
+  private async retrieveGrantsFromClient(issuerState: string): Promise<{ clientId: string; grants: Grant }> {
+    const credentialOfferState: CredentialOfferState | undefined = await this.retrieveCredentialOffer(issuerState)
+    const clientId = credentialOfferState?.clientId
+    const grants = credentialOfferState?.credentialOffer?.grants
+    if (!clientId) {
+      throw new Error(UNDEFINED_CLIENT_ID)
```

Comment:
Why are we throwing an error when the optional clientId is not present? It is the first call in the issue method, so a pre-authorized code flow without clientId will fail

Reply:
Correct! Removed the error.